python/tank/bootstrap/resolver.py

```diff
@@ -295,8 +296,60 @@ def _get_pipeline_configurations_for_project(self, pipeline_config_name, current
         )
 
         for pc in pipeline_configs:
+            # We'll need to provide a descriptor object for the config if
+            # possible. Note that it's possible that we'll be returning a
+            # None for the config descriptor. It's up to other filtering
+            # operations to remove those if desired.
+            #
+            # As in resolve_shotgun_configuration, the order of precedence
+            # is as follows:
+            #
+            # 1. windows/linux/mac path
+            # 2. descriptor
+            # 3. sg_descriptor
             path = ShotgunPath.from_shotgun_dict(pc)
 
+            try:
+                current_os_path = path.current_os
```

Comment:
In other words, personnally I'd yank the try/except out of there. As an added bonus the code will be easier on the eyes.